### PR TITLE
Update `cosmiconfig` to v6

### PR DESCRIPTION
**Pull request by fisker**
fixes #6810

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/config/resolve-config.js [hunk 1]
```diff
@@ -47,8 +48,8 @@ const getExplorerMemoized = mem(opts => {
     }
   });
 
-  const load = opts.sync ? explorer.loadSync : explorer.load;
-  const search = opts.sync ? explorer.searchSync : explorer.search;
+  const load = explorer.load;
+  const search = explorer.search;
```
Comment (evilebottnawi) [inline src/config/resolve-config.js]: We can simplify this - `const { load, search } = explorer;`

Comment (fisker) [reply src/config/resolve-config.js]: sure

Comment (fisker) [reply src/config/resolve-config.js]: we don't even need this const, use `explorer.load` directly

**Comment by fisker** [comment]
must be async function

**Comment by fisker** [comment]
includes....

**Comment by fisker** [comment]
should do something with the build script

**Comment by fisker** [comment]
another funny story, bug is from `import-fresh` , it clean require cache from parent module, make sense right? because calling require is from parent module no need clean `import-fresh` self, but when we bundle this module, there will be no parent, haha. too tired to confirm this.

https://github.com/fisker/prettier/pull/151

last two commits, going to sleep now, 2am...

**Comment by fisker** [comment]
@brodybits

**Comment by evilebottnawi** [comment]
@fisker I can take care of updating if you have problems

**Comment by brodybits** [comment]
I just tried `yarn prepare-release` and got `ReferenceError: regeneratorRuntime is not defined` on `tests_integration/__tests__/config-resolution.js`.

I think we should be able to resolve it by using `babel-plugin-transform-async-to-promises`, just like you did for the globby@10 update in PR #6776. I think we would need to keep this for browser support, even if we drop Node.js pre-8.0 support.

I would not personally favor pushing this update into the upcoming Prettier 1.19 release. Definitely a good job, though.

**Comment by fisker** [comment]
@brodybits i know that, success build bundle also not work

**Comment by fisker** [comment]
@evilebottnawi go for it

**Comment by fisker** [comment]
changed base to `next` branch

**Comment by evilebottnawi** [comment]
@fisker why next? Here breaking change?

**Comment by fisker** [comment]
@evilebottnawi 

Can't make bundle work on node@4 

- async functions need `babel-plugin-transform-async-to-promises` or `regenerate-runtime`,
- `util.promisify` requires another shim, like `pify`
- also need transform `includes` to `indexOf`
- most import, `require-fresh` not play well on node 4, hard to find a solution

not worse to dig

**Comment by evilebottnawi** [comment]
Okey, let's do it for next

**Pull request merged**